### DISCUSSION TITLE
Building Ruby extensions in the runtime

### DISCUSSION THREAD

**Discussion by andrealeone** [vercel/vercel#4466] [Help]
G'day,  
  
While deploying a cloud function with the [official Ruby runtime](https://vercel.com/docs/runtimes#official-runtimes/ruby),  
I get warnings like the following:
```
Ignoring nokogiri-1.10.9 because its extensions are not built. Try: gem pristine nokogiri --version 1.10.9
```
Since their native extensions have not been built, the gems are indeed missing,  
leading to this error:
```
Error raised from handler method
{
  "errorMessage": "Could not find nokogiri-1.10.9 in any of the sources",
  "errorType": "Function<Bundler::GemNotFound>",
  "stackTrace": [...]
}
```
<br>
Did I miss something in the configuration or is this intended in the current alpha state of the Ruby runtime? If so, do you plan to support native extensions?
<br>
<br>
Thank you.

**Comment by coetry** [comment]
Hi @andrealeone we can't realistically support every single native extension required, but can make certain extensions available if there is enough demand for it. Nokogiri is a really common Ruby library that depends on `libxml`, which we should be able to support. @styfle can we make that happen? 

Also @andrealeone can you share with us a repo of your code so we can run some tests?
